test/unit/test_host.rb

```diff
@@ -65,7 +65,6 @@ def test_assert_hosts_hash_equally
     end
 
     def test_assert_hosts_compare_equal
-      assert Host.new('example.com') == Host.new('example.com')
```

Comment:
I missed this in the review, good catch @robd - This SO question has a nice summary, and links the `Object` docs - http://stackoverflow.com/a/7157051/119669 - if we explicitly overwrite `==` (I'd have to check) then we need the test, if we don't, I don't think we do.